src/GameLogic/DataStructures/Entities.ts

```diff
@@ -22,8 +22,18 @@ export class Entities {
 	}
 
 	public getPlayer(): Protagonist | undefined {
-		// @todo Once we have projections this will have to be updated to return the correct player
-		return this.getFirstEntityOfType(EntityType.Protagonist) as Protagonist;
+		const protagonists = this.getEntitiesOfType(EntityType.Protagonist) as Protagonist[];
+		let result = undefined;
```

Comment:
I'd prefer a more functional solution, use `filter()` function to return all Protagonists with the necessary property, and if the result is >1 throw the error, otherwise return the first element

Reply:
This is a much better approach